Sources/_StringProcessing/Algorithms/Matching/Matches.swift

```diff
@@ -185,14 +185,26 @@ extension BidirectionalCollection {
 
 @available(SwiftStdlib 5.7, *)
 struct RegexMatchesCollection<Output> {
-  let input: Substring
+  let input: String
```

Comment:
I found it a lot clearer to separate them out as early as possible — otherwise you can have internal methods calling into others with things like `input[start...]`, which adjusts the wrong boundary.